ultralytics_yolov5.md

```diff
@@ -0,0 +1,180 @@
+```
+
+Note that cloning the [https://github.com/ultralytics/yolov5](https://github.com/ultralytics/yolov5) repository is **not** required.
+
+
+## Load YOLOv5 From PyTorch Hub
+
+### Simple Example
+
+This simple example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes a PIL image for inference. `'yolov5s'` is the lightest and fastest YOLOv5 model. Other available models are `'yolov5m'`, `'yolov5l'`, and `'yolov5x'`, the largest and most accurate model. For details on all models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints).
+```python
+import torch
+from PIL import Image
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/zidane.jpg', 'zidane.jpg')
+img = Image.open('zidane.jpg')  # PIL image
+
+# Inference
+prediction = model(img)  # includes NMS
+```
+
+### Detailed Example
+
+To load YOLOv5 from PyTorch Hub for **batched inference** with PIL, OpenCV, Numpy or PyTorch inputs, with results plotted, saved and printed to screen:
+```python
+import cv2
+import numpy as np
+import torch
+from PIL import Image, ImageDraw
+
+# Model
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).fuse().eval()  # yolov5s.pt
+model = model.autoshape()  # for autoshaping of PIL/cv2/np inputs and NMS
+
+# Images
+for f in ['zidane.jpg', 'bus.jpg']:  # download 2 images
+    torch.hub.download_url_to_file('https://github.com/ultralytics/yolov5/releases/download/v1.0/' + f, f)
+img1 = Image.open('zidane.jpg')  # PIL image
+img2 = cv2.imread('bus.jpg')[:, :, ::-1]  # OpenCV image (BGR to RGB)
+img3 = np.zeros((640, 1280, 3))  # numpy array
+imgs = [img1, img2, img3]  # batched list of images
+
+# Inference
+with torch.no_grad():
+    prediction = model(imgs, size=640)  # includes NMS
+
+# Plot
+for i, (img, pred) in enumerate(zip(imgs, prediction)):
+    str = 'Image %g/%g: %gx%g ' % (i + 1, len(imgs), *img.shape[:2])
+    img = Image.fromarray(img.astype(np.uint8)) if isinstance(img, np.ndarray) else img  # from np
+    if pred is not None:
+        for c in pred[:, -1].unique():
+            n = (pred[:, -1] == c).sum()  # detections per class
+            str += '%g %ss, ' % (n, model.names[int(c)])  # add to string
+        for *box, conf, cls in pred:  # xyxy, confidence, class
+            label = model.names[int(cls)] if hasattr(model, 'names') else 'class_%g' % cls
+            # str += '%s %.2f, ' % (label, conf)  # label
+            ImageDraw.Draw(img).rectangle(box, width=3)  # plot
+    img.save('results%g.jpg' % i)  # save
+    print(str + 'Done.')
+```
+
+<img src="https://user-images.githubusercontent.com/26833433/96368758-114c3a00-1156-11eb-9ede-a03f9aff42e4.jpg" width="550">
```

Comment:
remove these two inlined images. they are not needed, as when the user opens the hub page in a notebook and runs them, they will see the output

Reply:
Done.